cmd/sops/subcommand/groups/delete.go

```diff
@@ -0,0 +1,61 @@
+package groups
+
+import (
+	"os"
+
+	"go.mozilla.org/sops"
+	"go.mozilla.org/sops/cmd/sops/common"
+	"go.mozilla.org/sops/keyservice"
+)
+
+type DeleteOpts struct {
+	InputPath   string
+	InputStore  sops.Store
+	OutputStore sops.Store
+	Group       uint
+	GroupQuorum int
+	InPlace     bool
+	KeyServices []keyservice.KeyServiceClient
+}
+
+func min(a, b int) int {
+	if a > b {
+		return b
+	}
+	return a
+}
+
+func Delete(opts DeleteOpts) error {
+	tree, err := common.LoadEncryptedFile(opts.InputStore, opts.InputPath)
+	if err != nil {
+		return err
+	}
+	dataKey, err := tree.Metadata.GetDataKeyWithKeyServices(opts.KeyServices)
+	if err != nil {
+		return err
+	}
+	tree.Metadata.KeyGroups = append(tree.Metadata.KeyGroups[:opts.Group], tree.Metadata.KeyGroups[opts.Group+1:]...)
+
+	if opts.GroupQuorum != 0 {
+		tree.Metadata.ShamirQuorum = opts.GroupQuorum
+	}
+	// The quorum should always be smaller or equal to the number of key groups
+	tree.Metadata.ShamirQuorum = min(tree.Metadata.ShamirQuorum, len(tree.Metadata.KeyGroups))
```

Comment:
Will this silently change the quorum of a file should the user delete a keygroup that previously satisfied that quorum?
If so, I wonder if that's a good idea. I think we should make sure quorum modifications are intentional so users don't inadvertently reduce the security of their secrets files.

I would propose:
1. if deleting a key group results in a quorum that can no longer be satisfied, refuse the change and display an error
2. require users to update the quorum at the same time as deleting a key group for the operation to succeed: `sops groups --delete group1 --update-quorum 2`, or something along those lines

Reply:
Yeah, I think this might be better security-wise. The actual command is `sops groups delete --file file.yaml --shamir-secret-sharing-threshold 2 1`